framework/mongo/__init__.py

```diff
@@ -1,10 +1,41 @@
 # -*- coding: utf-8 -*-
 
-from modularodm import FlaskStoredObject as StoredObject
+from flask import request
+from modularodm.storedobject import StoredObject as GenericStoredObject
+from modularodm.ext.concurrency import with_proxies, proxied_members
 
 from bson import ObjectId
 from .handlers import client, database, set_up_storage
 
+
+from api.base.api_globals import api_globals
+
+
+class DummyRequest(object):
+    pass
+dummy_request = DummyRequest()
+
+
+def get_cache_key():
+    """
+    Fetch a request key from either a Django or Flask request. Fall back on a process-global dummy object
+    if we are not in either type of request
+    """
+    # TODO: This is ugly use of exceptions; is there a better way to track whether in a given type of request?
+    try:
+        return request._get_current_object()
+    except RuntimeError:  # Not in a flask request context
+        if hasattr(api_globals, 'request') and api_globals.request is not None:
```

Comment:
When removing the hasattr check...
```diff
diff --git a/framework/mongo/__init__.py b/framework/mongo/__init__.py
index 2ff142b..395ac83 100644
--- a/framework/mongo/__init__.py
+++ b/framework/mongo/__init__.py
@@ -25,7 +25,7 @@ def get_cache_key():
     try:
         return request._get_current_object()
     except RuntimeError:  # Not in a flask request context
-        if hasattr(api_globals, 'request') and api_globals.request is not None:
+        if api_globals.request is not None:
             return api_globals.request
         else:  # Not in a Django request
             return dummy_request
```

I get the following error when trying to start the apiserver (weird print behavior probably indicates multiple threads running):
```python
Unhandled exception in threaPerforming system checks...
d
 started by <function wrapper at 0x106c13848>
TrSystem check identified no issues (0 silenced).
aceback (most recent callJuly 02, 2015 - 13:57:09
 last):
  File "/Users/andyboughton/code/cos/osf.io/envDjango version 1.8, using settings 'api.base.settings'
/lib/python2.7/site-packages/django/utils/autoreload.pStarting development server at http://127.0.0.1:8000/
y", line 223, in wrapper
    fn(Quit the server with CONTROL-C.
*args, **kwargs)
  File "/Users/andyboughton/code/cos/osf.io/env/lib/python2.7/site-packages/django/core/management/commands/runserver.py", line 137, in inner_run
    handler = self.get_handler(*args, **options)
  File "/Users/andyboughton/code/cos/osf.io/env/lib/python2.7/site-packages/django/contrib/staticfiles/management/commands/runserver.py", line 24, in get_handler
    handler = super(Command, self).get_handler(*args, **options)
  File "/Users/andyboughton/code/cos/osf.io/env/lib/python2.7/site-packages/django/core/management/commands/runserver.py", line 55, in get_handler
    return get_internal_wsgi_application()
  File "/Users/andyboughton/code/cos/osf.io/env/lib/python2.7/site-packages/django/core/servers/basehttp.py", line 50, in get_internal_wsgi_application
    return import_string(app_path)
  File "/Users/andyboughton/code/cos/osf.io/env/lib/python2.7/site-packages/django/utils/module_loading.py", line 26, in import_string
    module = import_module(module_path)
  File "/usr/local/Cellar/python/2.7.9/Frameworks/Python.framework/Versions/2.7/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/Users/andyboughton/code/cos/osf.io/api/base/wsgi.py", line 16, in <module>
    init_app(set_backends=True, routes=False, attach_request_handlers=False)
  File "/Users/andyboughton/code/cos/osf.io/website/app.py", line 135, in init_app
    ensure_schemas()
  File "/Users/andyboughton/code/cos/osf.io/website/project/model.py", line 116, in ensure_schemas
    schema_obj.save()
  File "/Users/andyboughton/code/cos/osf.io/env/lib/python2.7/site-packages/modularodm/storedobject.py", line 110, in wrapped
    return func(*args, **kwargs)
  File "/Users/andyboughton/code/cos/osf.io/env/lib/python2.7/site-packages/modularodm/storedobject.py", line 82, in wrapped
    return func(this, *args, **kwargs)
  File "/Users/andyboughton/code/cos/osf.io/env/lib/python2.7/site-packages/modularodm/storedobject.py", line 783, in save
    cached_data = self._get_cached_data(self._stored_key)
  File "/Users/andyboughton/code/cos/osf.io/env/lib/python2.7/site-packages/modularodm/storedobject.py", line 457, in _get_cached_data
    return cls._cache.get(cls._name, key)
  File "/Users/andyboughton/code/cos/osf.io/env/lib/python2.7/site-packages/werkzeug/local.py", line 338, in __getattr__
    return getattr(self._get_current_object(), name)
  File "/Users/andyboughton/code/cos/osf.io/env/lib/python2.7/site-packages/werkzeug/local.py", line 297, in _get_current_object
    return self.__local()
  File "/Users/andyboughton/code/cos/osf.io/env/lib/python2.7/site-packages/modularodm/ext/concurrency.py", line 39, in local
    key = get_key()
  File "/Users/andyboughton/code/cos/osf.io/framework/mongo/__init__.py", line 28, in get_cache_key
    if api_globals.request is not None:
AttributeError: 'thread._local' object has no attribute 'request'
```